rfcs/react-components/convergence/reusing-avatar-in-components.md

```diff
@@ -0,0 +1,265 @@
+# RFC: Reusing Avatar in other components
+
+@ling1726
+
+## Summary
+
+This RFC enumerates the possible ways to handle:
+
+- Avatar default props
+- Avatar default sizing
+
+When the component is reused within other v9 components. There is a broader question of how to
+internally reuse components in general, hopefully resolving the problem described in this RFC
+will help us make easier decisions for component reuse in the future.
+
+## Background
+
+Fluent UI v9 has design requirements on each of its components. These requirements are compiled from product
+usage. This is a good thing since our components can be easily used to power products. However one of the consequences
+of stricter design requirements it the resuse of our own components. Sometimes the component reusage can involve
+different defaults. How can we handle this in React Library?
+
+## Problem statement
+
+Let's use `Avatar` as an example. We have concrete problems here that involve the `Table` and `Persona` components.
+Both of these components use `Avatar` and require a default size when used in those components.
+
+```tsx
+<Table size="smaller">
+  <TableBody>
+    <TableRow>
+      <TableCell>
+        {/* The Avatar should have size 20 to follow the Table size */}
+        <TableCellLayout media={<Avatar />}>Main content</TableCellLayout>
+      </TabeCell>
+    </TableRow>
+  </TableBody>
+</Table>
+```
+
+The default size for the `Avatar` component is `32`. This means that the component that is reusing the avatar needs
+to do some extra work to make sure that the avatar follows design guidance.
+
+## Detailed Design or Proposal
+
+### Userland problem
+
+We could treat this as a userland problem, and require that apps/features follow design guidance when using an
+avatar with these components.
+
+```tsx
+<Table size="smaller">
+  <TableBody>
+    <TableRow>
+      <TableCell>
+        {/* Explicitly set the size */}
+        <TableCellLayout media={<Avatar size={20} />}>Main content</TableCellLayout>
+      </TabeCell>
+    </TableRow>
+  </TableBody>
+</Table>
+```
+
+#### Pros
+
+- No extra logic in Fluent code
+
+#### Cons
+
+- Users must know about design guidance
+- Cannot ship design updates i.e. users will need to upgrade all usages if design decides that "64px" will be new size
+- Extra work for users
+
+### Style override
```

Comment:
Have we considered the usage of CSS variables for the style overrides proposal? This solves the "Does not handle props" con. If the size prop of the avatar adds styles with the CSS variable, it'll have priority over the overriden style by the Table.

A con to this approach is to document well the CSS variable.